shared/dataflow/codeql/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -522,14 +502,155 @@ module MakeImplCommon<LocationSig Location, InputSig<Location> Lang> {
       )
     }
 
+    private module CallSetsInput implements MkSetsInp {
```

Comment:
Something like that could work as well, but I chose the current version to avoid wrapping all of the `DispatchSet` edges in 'some'-wrappers. Also, it becomes quite tricky to get right if we stuff everything together like that - your version above e.g. fails to specify which calls have no dispatch, so an option-type doesn't carry enough information. For a given call affected by the context, there's either a non-empty set of dispatch edges or an empty set, but that empty set cannot be represented by a none value, because then we don't know what call it refers to.

Reply:
Right, the `Option` would have to apply to the `Callable` in `TCallEdge`. Let's leave it as-is then. I think this section deserves a comment then about why we need both `CallSets` and `DispatchSets`.